internal/blockchain/doc.go

```diff
@@ -1,17 +1,36 @@
 /*
-Package blockchain provides two implementations of the fast-sync protocol.
+Package blockchain implements two versions of a reactor Service that are
+responsible for block propagation and gossip between peers. This mechanism is
+more formally known as fast-sync.
 
-- v0 was the very first implementation. it's battle tested, but does not have a
-lot of test coverage.
-- v2 is the newest implementation, with a focus on testability and readability.
+In order for a validator to successfully participate in consensus, it must have
```

Comment:
This is a bit of nit, but it's not just validators participating in consensus. All full nodes can

Reply:
Yeah, I just didn't feel like making that distinction 😆